packages/components/data-resource/s3-endpoint-type.tsx

```diff
@@ -30,14 +48,91 @@ type S3EndpointTypeProps = {
   namespace: string;
 };
 
+const isValidNS = (projName: string) => {
+  const isValid = EXCLUDED_PREFIX.reduce((acc, cur) => {
+    return acc && !projName.startsWith(cur);
+  }, true);
+  return isValid && !EXCLUDED_NS.includes(projName);
+};
+
+const { namespace, ...secretResourceWithoutNamespace } = secretResource;
+const getUpdatedSecretResource = (namespace: string) => {
+  const updatedSecretResource: WatchK8sResource = {
+    ...secretResourceWithoutNamespace,
+    namespace,
+  };
+  return updatedSecretResource;
+};
+
 const regionDropdownOptions: (t: TFunction) => JSX.Element[] = (t) =>
   _.map(awsRegionItems, (v, _) => <SelectOption key={v} value={v} />);
+const getSecondaryTextForSecrets = (resource: K8sResourceCommon): string => {
+  return resource?.metadata?.namespace;
+};
 
 export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
   const { t } = useTranslation();
+  const { provider, state, dispatch, type } = props;
 
   const [showSecret, setShowSecret] = React.useState(true);
-  const { provider, state, dispatch, type } = props;
+  const [canCreateSecret, setCanCreateSecret] = React.useState(true);
+  const [canReadSecret, setCanReadSecret] = React.useState(true);
+
+  const [isClusterAdmin, isClusterAdminLoading] = useAccessReview({
+    group: SecretModel.apiGroup,
+    resource: SecretModel.plural,
+    verb: 'create' as K8sVerb,
+    namespace: DATA_FEDERATION_NAMESPACE,
+  });
+
+  React.useEffect(() => {
+    if (
+      !isClusterAdminLoading &&
+      !isClusterAdmin &&
+      state.secretNamespace === initialStateCreateForm.secretNamespace
+    ) {
+      dispatch({
+        type: createFormAction.SET_SECRET_NAMESPACE,
+        value: DEDICATED_ADMIN,
+      });
+    }
+  }, [dispatch, isClusterAdmin, isClusterAdminLoading, state.secretNamespace]);
```

Comment:
this will create bugs...
start with empty namespace in initialStateCreateForm.secretNamespace = ''.
```suggestion
  React.useEffect(() => {
    if (!isClusterAdminLoading && state.secretNamespace === '')
    ) {
      dispatch({
        type: createFormAction.SET_SECRET_NAMESPACE,
        value: isClusterAdmin ? REDHAT_DATA_FEDERATION : DEDICATED_ADMIN,
      });
    }
  }, [dispatch, isClusterAdmin, isClusterAdminLoading, state.secretNamespace]);
```